### ISSUE TITLE
Better SoC for LRS + make easier to use

### ISSUE THREAD

**Issue by dpressel** [dpressel/mead-baseline#653]
Many of the learning rate schedulers have a decay number of steps, but this is sometimes not super obvious how to provide, as the epochs are passed to the trainer, which means that it cannot be injected into the LRS.  We get around this with mead by having a `calc_lr_params()` function that injects the steps / epoch computations but this isnt really an obvious place for this.

Would make more sense for the trainer to pass in the steps / epochs and max training epochs into the `create_trainer()`, which could cascade into the actual LRS, and that could take in some defaults that would allow it to, e.g. decay to 0 at end of training?
